Add scaladget library

mathieuleclaire opened the pull request:
Added scaladget library. Hope it could help.

gzm0 commented:
Thanks for the contribution. Please change the commit subject to "Add scaladget library". "Update. index.md" is not very useful in a commit list (and the commit body does not show).

mathieuleclaire commented:
Done !

gzm0 commented:
Ah, sorry... I meant the one in the commit itself. The GitHub PR title will only appear in the merge commit (but indeed, it needed changing too).

You can amend the commit (`git commit --amend`) and then force push (`git push -f origin master`).

Also (in the future), please avoid making PRs from your master branch (if you continue working on it, the PR changes, which is rarely what you want).

mathieuleclaire commented:
OK, I did it; everything was fine locally, but I don't see the propagation of my modification of the github page, so that I think my amend failed. I had to merge #d4c0ad783a27ae139f93594cd066c7e63da2c7fb in between, peharps it is why it failed. Do you know how to fix it ?

gzm0 commented:
Yep... The merge messed things up.

Do the following: (warning: deletes all other changes to your working copy)

```
git reset --hard 7a36d3ee5c6ac6e1ced631ceea2e503948a0e1a0
git commit --amend
# Change commit message in the editor
git push -f origin master
```

If any of the commands fail: Don't run what they say, tell me the error.

mathieuleclaire commented:
I tried this already and I got (and I still get) a "Everything up-to-date" after the push, meaning it has no effect.

gzm0 commented:
That is very strange. Your `origin/master` (assuming it is configured to be on your GitHub repo) is still at 1cb845a094399ef2504c502a7fb803379e60b92b.

Could you try:

```
git push -f origin 7a36d3ee5c6ac6e1ced631ceea2e503948a0e1a0:master
```

Also, could you show me the output of `git show` and `git log` on your local WC?

Thanks

mathieuleclaire commented:
The command worked, but the commit on github still has the old title ... I tried to reset again and redo the amend and the git push -f origin 7a36d3ee5c6ac6e1ced631ceea2e503948a0e1a0:master, but I got again the "Everything up-to-date" :(. 

git show gives:
```sh
commit 9e4f310b93a8ee642ad5351107e5f48be30780bb
Author: Mathieu <mathieu.leclaire@iscpif.fr>
Date:   Thu Sep 3 08:33:49 2015 +0200

    Add scaladget library
    
    Added scaladget library

diff --git a/index.md b/index.md
index ac76271..c6d6eb9 100644
--- a/index.md
+++ b/index.md
@@ -84,6 +84,7 @@ and then "Create a Pull Request".
 *   [threejs-facade](https://github.com/antonkulaga/threejs-facade): static types for [Three.js 3D library](https://threejs.org)
 *   [codemirror-facade](https://github.com/antonkulaga/codemirror-facade): static types for [Codemirror code editor](http://codemirror.net)
 *   [selectize-facade](https://github.com/antonkulaga/selectize-facade): static types for [Selectize](http://brianreavis.github.io/selectize.js/)
+*   [scaladget](https://github.com/mathieuleclaire/scaladget): static types for [D3.js](d3js.org), [Ace](http://ace.c9.io), [Tooltipster](http://iamceege.github.io/tooltipster/),  [Bootstrap.js](http://getbootstrap.com/) (and an API for the latter)
 
 
 ### Testing frameworks
```

and git log
```sh
commit 9e4f310b93a8ee642ad5351107e5f48be30780bb
Author: Mathieu <mathieu.leclaire@iscpif.fr>
Date:   Thu Sep 3 08:33:49 2015 +0200

    Add scaladget library
    
    Added scaladget library
```
which seems to be good, but on github, I do not have the same title: https://github.com/mathieuleclaire/scala-js-website/commits/master.
Weird !

gzm0 commented:
Ah, OK. But now on GitHub, you have the old commit. Now do `git push -f origin master` again. (Or if it doesn't work: `git push -f origin 9e4f310b93a8ee642ad5351107e5f48be30780bb:master`).

mathieuleclaire commented:
doing this made me merge again with 7a36d3ee5c6ac6e1ced631ceea2e503948a0e1a0. Are we in a github black hole :) !?

gzm0 commented:
I don't understand why pushing merges anything... :( Does specifying the commit explicitly work?

mathieuleclaire commented:
what do you mean ?

gzm0 commented:
Does this work: `git push -f origin 9e4f310b93a8ee642ad5351107e5f48be30780bb:master`?

mathieuleclaire commented:
Yes ! Sorry, I didn't see that a new commit ID had been generated. Thanks for helping ! I will take care next time to comment properly my commit titles.

gzm0 commented on index.md:
```diff
@@ -84,6 +84,7 @@ and then "Create a Pull Request".
 *   [threejs-facade](https://github.com/antonkulaga/threejs-facade): static types for [Three.js 3D library](https://threejs.org)
 *   [codemirror-facade](https://github.com/antonkulaga/codemirror-facade): static types for [Codemirror code editor](http://codemirror.net)
 *   [selectize-facade](https://github.com/antonkulaga/selectize-facade): static types for [Selectize](http://brianreavis.github.io/selectize.js/)
+*   [scaladget](https://github.com/mathieuleclaire/scaladget): static types for [D3.js](d3js.org), [Ace](http://ace.c9.io), [Tooltipster](http://iamceege.github.io/tooltipster/),  [Bootstrap.js](http://getbootstrap.com/) (and an API for the latter)
```
`d3js.org` is missing `http://`. This will not work correctly.

gzm0 commented:
No worries :) This will give you good git training. (If you amend, the commit ID *must* change, it is the SHA1 of the commits contents).

So... Now you have the line "Add scaladget library" twice in the commit message. That is not really a big deal so I'd merge it anyways. But there is an issue in one of the links (I commented individually).

gzm0 commented:
So while you are at it, you can also fix the commit message :)

mathieuleclaire commented:
How can you modify the commit itself ? I just succeeded in changing title and message but my new verison of index.md (with missing http) has not been taking into account. I just tried the same thing than before.

gzm0 commented:
You need to stage the changes (using `git add`). When you then amend, they will be put into the new commit.

mathieuleclaire commented:
Done ! Thanks.

sjrd commented:
LGTM

gzm0 commented:
Thanks for bearing with us on this one @mathieuleclaire

mathieuleclaire commented:
No worries, I didn't know the amend option; I know it very well now :) !
